src/driver_controller.cpp

```diff
@@ -0,0 +1,23 @@
+#include "drivers/driver_controller.hpp"
+
+#include "drivers/led_display.hpp"
+#include "drivers/timer.hpp"
+
+namespace edge::drivers {
+void do_work()
+{
+    led_display.display_pixels_once();
+}
+
+etl::optional<int> handle_command(DriverType type, int arg1, int arg2, int arg3)
+{
+    switch (type) {
```

Comment:
can we add a throw on default case?

Reply:
its embedded, we have no throws lol

i can add an abort because the default case should never happen unless we forget to add all the switch cases

Reply:
... my fault lmao

Reply:
ima resolve this because it gives a compiler warning if we don't handle cases, i think thats enough

Reply:
@echavemann there was actually a very interesting talk on exceptions in embedded at cppcon which i recommend watching when they post it